src/test/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBaseTest.java

```diff
@@ -422,4 +464,34 @@ void testGap(MtAmazonDynamoDbBase mtDynamoDb, MtAmazonDynamoDbStreams mtDynamoDb
             deleteMtTables(mtDynamoDb);
         }
     }
+
+    /**
+     * Tests that the describeStreamCache is utilized when describeStream is called in StreamsBase.
+     */
+    @ParameterizedTest
+    @ArgumentsSource(Args.class)
+    void testDescribeStreamCache(MtAmazonDynamoDbBase mtDynamoDb, MtAmazonDynamoDbStreams mtDynamoDbStreams,
+                                 CachingAmazonDynamoDbStreams cachingStreams) {
+
+        // Some of the parameterized inputs don't use a CachingAmazonDynamoDbStreams instance
+        if (cachingStreams == null) {
+            return;
+        }
+
+        try {
+            createTenantTables(mtDynamoDb);
+            List<Stream> streams = mtDynamoDbStreams.listStreams(new ListStreamsRequest()).getStreams();
+
+            for (Stream stream : streams) {
+                String key = stream.getStreamArn();
+                DescribeStreamResult expectedResult =
+                    new DescribeStreamResult().withStreamDescription(new StreamDescription().withStreamArn(key));
+
+                assertDescribeStreamCache(mtDynamoDbStreams, cachingStreams, stream, false, expectedResult);
```

Comment:
This fails when run with all the tests, not when the test class is run.   I'm missing how the shardIterator might be empty in this case.  (looks like another test suite is interfering but I wasn't able to find which one). 
[ERROR] Failures:
[ERROR]   MtAmazonDynamoDbStreamsBaseTest.testDescribeStreamCache:490->assertDescribeStreamCache:250 expected: <true> but was: <false>
[ERROR]   MtAmazonDynamoDbStreamsBaseTest.testDescribeStreamCache:490->assertDescribeStreamCache:250 expected: <true> but was: <false>
[ERROR]   MtAmazonDynamoDbStreamsByTableTest>MtAmazonDynamoDbStreamsBaseTest.testDescribeStreamCache:490->MtAmazonDynamoDbStreamsBaseTest.assertDescribeStreamCache:250 expected: <true> but was: <false>
[ERROR]   MtAmazonDynamoDbStreamsByTableTest>MtAmazonDynamoDbStreamsBaseTest.testDescribeStreamCache:490->MtAmazonDynamoDbStreamsBaseTest.assertDescribeStreamCache:250 expected: <true> but was: <false>

Reply:
fixed in 8af378f4530cba778d8315f80d5b910073904e7a